packages/bot-skeleton/src/scratch/hooks/block_svg.js

```diff
@@ -223,6 +223,13 @@ Blockly.BlockSvg.prototype.setCollapsed = function (collapsed) {
         return;
     }
 
+    // Firefox fix for Blockly widthcache bug
+    if (navigator.userAgent.search('Firefox') > 0) {
+        setTimeout(() => {
+            this.workspace.getAllFields().forEach(field => field.forceRerender());
+        }, 0);
```

Comment:
Why do we need a `setTimeout` if we're not giving it a time duration?

Reply:
we need to make it asynchronous for its proper work.  Without it it works from time to time so we have to ensure that it will run after all other functions

Reply:
What do you mean?

> Without it it works from time to time

Reply:
it was rerenderig block before the change of its shape, as I understood

Reply:
as bug appears not every time, I'm not sure if it was helping at all

Reply:
Okay, please add a comment there as to why we want to leave the duration as 0. 

e.g.:
```suggestion
        setTimeout(() => {
            this.workspace.getAllFields().forEach(field => field.forceRerender());
        }, 0); // Time duration must be 0
```